test/helpers/utils.go

```diff
@@ -122,15 +122,33 @@ func WithTimeout(body func() bool, msg string, config *TimeoutConfig) error {
 	done := time.After(config.Timeout * time.Second)
 	ticker := time.NewTicker(config.Ticker * time.Second)
 	defer ticker.Stop()
-	if body() {
-		return nil
+	start := make(chan bool)
+	status := make(chan bool)
+
+	// Send the first execution of the body func
+	go func() { start <- true }()
+
+	// If the function take more than the config.Ticker the function will be
+	// not executed.
+	bodyLock := false
+	WithLock := func(body func() bool) {
+		if bodyLock == true {
```

Comment:
the bool type is not safe to use by multiple goroutines you need to use a locking mechanism to protect this variable.

Reply:
Could you share with me why it's not safe? Could you describe/share any snippet where is not safe?